packages/main-api/src/controllers/v1/spaces.rs

```diff
@@ -843,16 +851,202 @@ impl SpaceController {
+
+        // === DELETE ELEARNING ===
+        let elearnings = Elearning::query_builder()
+            .space_id_equals(space_id)
+            .query()
+            .map(Elearning::from)
+            .fetch_all(&self.pool)
+            .await?;
+
+        for elearning in elearnings {
+            self.elearning_repo
+                .delete_with_tx(&mut *tx, elearning.id)
+                .await?;
+        }
+
+        // === DELETE SPACE MEMBERS ===
+        let members = SpaceMember::query_builder()
+            .space_id_equals(space_id)
+            .query()
+            .map(SpaceMember::from)
+            .fetch_all(&self.pool)
+            .await?;
+
+        for member in members {
+            self.space_member_repo
+                .delete_with_tx(&mut *tx, member.id)
+                .await?;
+        }
+
+        // === DELETE SPACE DRAFT ===
+        let drafts = SpaceDraft::query_builder()
+            .space_id_equals(space_id)
+            .query()
+            .map(SpaceDraft::from)
+            .fetch_all(&self.pool)
+            .await?;
+
+        for draft in drafts {
+            self.space_draft_repo
+                .delete_with_tx(&mut *tx, draft.id)
+                .await?;
+        }
+
+        // ===  DELETE SpaceLikeUser / SpaceShareUser  ===
+        let like_repo = SpaceLikeUser::get_repository(self.pool.clone());
+        let share_repo = SpaceShareUser::get_repository(self.pool.clone());
+
+        let likes = SpaceLikeUser::query_builder()
+            .space_id_equals(space_id)
+            .query()
+            .map(SpaceLikeUser::from)
+            .fetch_all(&self.pool)
+            .await?;
+
+        for like in likes {
+            like_repo.delete_with_tx(&mut *tx, like.id).await?;
+        }
+
+        let shares = SpaceShareUser::query_builder()
+            .space_id_equals(space_id)
+            .query()
+            .map(SpaceShareUser::from)
+            .fetch_all(&self.pool)
+            .await?;
+
+        for share in shares {
+            share_repo.delete_with_tx(&mut *tx, share.id).await?;
+        }
+
+        // === DELETE SPACE GROUPS ===
+        let groups = SpaceGroup::query_builder()
+            .space_id_equals(space_id)
+            .query()
+            .map(SpaceGroup::from)
+            .fetch_all(&self.pool)
+            .await?;
+        for group in groups {
+            self.space_group_repo
+                .delete_with_tx(&mut *tx, group.id)
+                .await?; // Changed to use self.space_group_repo
+        }
+
+        // === DELETE SURVEY RESPONSES ===
+        let response_repo = SurveyResponse::get_repository(self.pool.clone());
+        let responses = SurveyResponse::query_builder()
+            .space_id_equals(space_id)
+            .query()
+            .map(SurveyResponse::from)
+            .fetch_all(&self.pool)
+            .await?;
+        for resp in responses {
+            response_repo.delete_with_tx(&mut *tx, resp.id).await?;
+        }
```

Comment:
Don't delete spaces cascading other dependency.
Users must guarantee  space being empty in advance. @Jamesunekwuojo